pkg/systemd/activation/activation.go

```diff
@@ -0,0 +1,36 @@
+package activation
+
+import (
+  "os"
+  "os/exec"
+  "strings"
+  "syscall"
+  "strconv"
+)
+
+type Opt struct {
+  RunActivationHelperEnvKey  string   // needs to be set
+  TargetCmd                  []string // needs to be set
+}
+
+func ActivationHelper(opt Opt) error {
+  pid := os.Getpid()
+  os.Unsetenv(opt.RunActivationHelperEnvKey)
+  os.Setenv("LISTEN_PID", strconv.Itoa(pid))
+  argsv := opt.TargetCmd
+  execPath, err := selectExecutable(argsv[0])
+  if err != nil {
+    return err
+  }
+  if err = syscall.Exec(execPath, argsv, os.Environ()); err != nil {
+    return err
+  }
+  panic("should not reach here")
+}
+
+func selectExecutable(path string) (string, error) {
+  if strings.Contains(path, "/") {
+    return path, nil
+  }
+  return exec.LookPath(path)

```

Comment:
Can't we unconditionally use `exec.LookPath`?

Reply:
You are right. `if` branch is unnecessary.